SegmentEditorFloodFilling/SegmentEditorFloodFillingLib/SegmentEditorEffect.py

```diff
@@ -115,6 +116,7 @@ def masterVolumeNodeChanged(self):
   def setMRMLDefaults(self):
     self.scriptedEffect.setParameterDefault("IntensityTolerance", 10.0)
     self.scriptedEffect.setParameterDefault("NeighborhoodSizeMm", 1.0)
+    self.scriptedEffect.setParameterDefault("ROINodeID", "")
```

Comment:
Node IDs do not rename the same when you save and load the scene (because for example you may have the same node IDs in the scene that are used in a saved scene; so the node IDs are automatically renamed when you load the scene). You must use node references instead (SetNodeReferenceID/GetNodeReferenceID) when you want to keep a link from a node to another node.

Reply:
Thank you for you reply.

But I don't find Set/GetNodeReferenceID with an effect : 

```
seWidgetEditor = slicer.modules.SegmentEditorWidget.editor
seWidgetEditor.setActiveEffectByName("Flood filling")
ffEffect = seWidgetEditor.activeEffect()
```

Neither ffEffect nor ffEffect.self() expose such function. How do we get to these functions ?

Reply:
I'm getting close using

`self.scriptedEffect.parameterSetNode()`

Not yet fully functional. Will push a change soon.

Reply:
I've made the necessary changes, probably badly. Because when I set the ROI node in FloodFilling from [remote](https://github.com/chir-set/SlicerExtension-VMTK/blob/e623cec083223136436788d914ac0f39857cf801/FiducialCenterlineExtraction/FiducialCenterlineExtraction.py#L582
), it remains None. If I just run the same code a second time, it is correctly set. Can't figure out what's wrong.

Reply:
I think it's not your fault. GUI update is only triggered by slicer.vtkMRMLSegmentEditorNode.EffectParameterModified. Probably invoking such an event on the parameter node after updating the node reference will force the GUI to update.

Reply:
Calling ffEffect.updateGUIFromMRML() refreshes the UI after setting the ROI node. I suppose it's an acceptable solution. If so, I'll be running with that. Thanks.